scripts/o3de/o3de/register.py

```diff
@@ -493,6 +493,10 @@ def register_repo(json_data: dict,
     repo_sha256 = hashlib.sha256(url.encode())
     cache_file = manifest.get_o3de_cache_folder() / str(repo_sha256.hexdigest() + '.json')
 
+    git_tuple = utils.is_git_provider_uri(parsed_uri)
+    if git_tuple[0]:
+        parsed_uri = git_tuple[1](parsed_uri)
+
```

Comment:
```suggestion
    if utils.is_git_provider_uri(parsed_uri):
        parsed_uri = utils.get_git_provider_file_url(parsed_uri)
```

An alternative approach might look more like this:
```
   git_provider = utils.get_git_provider_from_uri(parsed_uri) # returns a provider object specific to GitHub, CodeCommit etc. based on the URI
  if git_provider:
     parsed_uri = git_provider.get_file_url(parsed_uri)
```